src/audio/module_adapter/module/modules.c

```diff
@@ -185,7 +182,7 @@ static int modules_prepare(struct processing_module *mod,
 	/* Call module specific prepare function if exists. */
 	if (mod->is_native_sof) {
 		struct module_interface *mod_in =
-					(struct module_interface *)mod->priv.module_adapter;
+					(struct module_interface *)mod->priv.ops;
```

Comment:
you don't need these type-casts any more and in fact they're wrong: `.ops` is `const`, and in these assignments `const` is dropped. Please, remove the type-cast and add `const` to `struct module_interface *mod_in`